src/entities/nativeCurrencyTypes.ts

```diff
@@ -0,0 +1,16 @@
+import nativeCurrencyReference from '../references/native-currencies.json';
+
+type NativeCurrencyKey = keyof typeof nativeCurrencyReference;
+type NativeCurrencyKeysMap = { [key in NativeCurrencyKey]: NativeCurrencyKey };
```

Comment:
There's one downside to this approach — trying to get the type of a single value, e.g. `typeof NativeCurrencyKeys.USD`, will resolve to the entire union of `'ETH' | 'USD' | 'EUR' | 'GDP'` etc. rather than just `'USD'`. If we rewrite this line to use the generic `Key` value, we maintain the relationship between keys and values:

```ts
type NativeCurrencyKeysMap = { [Key in NativeCurrencyKey]: Key };
```

This way, `typeof NativeCurrencyKeys.USD` will resolve to `'USD'` as expected.

I tried this out myself but it made the typing a bit more difficult when iterating over the object, so I swapped it out for a lodash `mapValues` call. Here's the file in its entirety. What do you think?

```ts
import { mapValues } from 'lodash';
import nativeCurrencyReference from '../references/native-currencies.json';

type NativeCurrencyKey = keyof typeof nativeCurrencyReference;
type NativeCurrencyKeysMap = { [Key in NativeCurrencyKey]: Key };

// We can't dynamically generate an enum from the JSON data, but we can
// generate this type-checked object, which is nicer than using strings directly.

/**
 * An enum of native currencies such as "USD" or "ETH".
 */
export const NativeCurrencyKeys = mapValues(
  nativeCurrencyReference,
  (_value, key) => key
) as NativeCurrencyKeysMap;
```

Reply:
I see, that makes sense! Thanks for the detailed note and code snippet. I made those changes to the definition and implementation of `NativeCurrencyKeys` and pushed. Bitrise is running now.